### DISCUSSION TITLE
How to deploy ink! contract?

### DISCUSSION THREAD

**Discussion by CocDap** [polkadot-api/polkadot-api#849] [Q&A] [answered]
I referred to this documentation: https://papi.how/ink#deploy-contract. However, I am unsure how to proceed with the actual deployment and have only mentioned to perform a `dry-run` check

**Comment by voliva** [comment, marked as answer]
Essentially sending a transaction to deploy the contract instead of through a runtime-API. The docs you linked suggest using `typedApi.tx.Contracts.instantiate_with_code`, but there are other calls that can be used to deploy one, each one with different parameters.

We don't currently have an example in the docs sending a transaction, but you can find one making a call to a contract in our [example file](https://github.com/polkadot-api/polkadot-api/blob/ccbb6b36b3022aa45dfe44badff393359b2a0b2b/examples/ink/index.ts#L146). I hope you can extrapolate this to do the contract deploy call.

We have designed the ink-client completely chain-agonstic, following the modular design of papi, but this means that it requires some extra work to use it with specific chains. 

We're now working on ink-sdk, a separate package built on top of papi and ink-client which will abstract over this and make it extremely simple to directly get the actual transaction to perform a deployment (as well as other actions).
